File: database/src/main/kotlin/com/cip/agdxapi/database/entities/CommonDataEntity.java
package com.cip.agdxapi.database.entities;


import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "common_data")
public class CommonDataEntity extends com.cip.agdxapi.database.entities.BaseEntity {

    @Column(name = "lat", nullable = false)
    public BigDecimal lat;

    @Column(name = "lon", nullable = false)
    public BigDecimal lon;

    @Column(name = "alt", nullable = false)
    public BigDecimal alt;

    @Column(name = "accuracy", nullable = false)
    public BigDecimal accuracy;

    @Column(name = "country_code", nullable = false)
    public String countryCode;

    @Column(name = "country_name", nullable = false)
    public String countryName;

    @Column(name = "location_level1")
    public String locationLevel1;

    @Column(name = "location_level2")
    public String locationLevel2;

    @Column(name = "location_level3")
    public String locationLevel3;

    @Column(name = "date_reported")
    public LocalDateTime dateReported;

    @Column(name = "date_updated")
    public LocalDateTime dateUpdated;

    @Column(name = "date_reviewed")
    public LocalDateTime dateReviewed;

    @Column(name = "ontology_name")
    public String ontologyName;

    @Column(name = "common_name")
    public String commonName;

    @Column(name = "scientific_name")
    public String scientificName;

    @Column(name = "area_sampled")
    public BigDecimal areaSampled;

    @Column(name = "affected_area")
    public BigDecimal affectedArea;

    @Column(name = "area_unit")
    public BigDecimal areaUnit;

    @Column(name = "sample_source")
    public String sampleSource;

    @Column(name = "sample_desc")
    public String sampleDesc;

    @Column(name = "sample_size")
    public Integer sampleSize;

    @Column(name = "affected_sample_size")
    public Integer affectedSampleSize;

    @Column(name = "severity")
    public Integer severity;

    /**
     * whether detected or undetected
     */
    @Column(name = "occurrence_status")
    public String occurrenceStatus;

    @Column(name = "comments")
    public String comments;

    @Column(name = "identification_method")
    public String identificationMethod;

    @Column(name = "data_collection_method")
    public String dataCollectionMethod;

    @Column(name = "visually_identified")
    public Boolean visuallyIdentified;

    @Column(name = "image_url")
    public String imageUrl;

    @Column(name = "identified_by")
    public String identifiedBy;

}

File: database/src/main/kotlin/com/cip/agdxapi/database/entities/CommonDataEntity.kt
package com.cip.agdxapi.database.entities

import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "common_data")
class CommonDataEntity : BaseEntity() {
    @Column(name = "lat", nullable = false)
    var lat: BigDecimal? = null

    @Column(name = "lon", nullable = false)
    var lon: BigDecimal? = null

    @Column(name = "alt", nullable = false)
    var alt: BigDecimal? = null

    @Column(name = "accuracy", nullable = false)
    var accuracy: BigDecimal? = null

    @Column(name = "country_code", nullable = false)
    var countryCode: String? = null

    @Column(name = "country_name", nullable = false)
    var countryName: String? = null

    @Column(name = "location_level1",columnDefinition = "TEXT")
    var locationLevel1: String? = null

    @Column(name = "location_level2",columnDefinition = "TEXT")
    var locationLevel2: String? = null

    @Column(name = "location_level3",columnDefinition = "TEXT")
    var locationLevel3: String? = null

    @Column(name = "date_reported")
    var dateReported: LocalDate? = null

    @Column(name = "date_updated")
    var dateUpdated: LocalDate? = null

    @Column(name = "date_reviewed")
    var dateReviewed: LocalDate? = null

    @Column(name = "ontology_name")
    var ontologyName: String? = null

    @Column(name = "common_name")
    var commonName: String? = null

    @Column(name = "scientific_name")
    var scientificName: String? = null

    @Column(name = "area_sampled")
    var areaSampled: BigDecimal? = null

    @Column(name = "affected_area")
    var affectedArea: BigDecimal? = null

    @Column(name = "area_unit")
    var areaUnit: String? = null

    @Column(name = "sample_source",columnDefinition = "TEXT")
    var sampleSource: String? = null

    @Column(name = "sample_desc",columnDefinition = "TEXT")
    var sampleDesc: String? = null

    @Column(name = "sample_size")
    var sampleSize: Int? = null

    @Column(name = "affected_sample_size")
    var affectedSampleSize: Int? = null

    @Column(name = "severity")
    var severity: Int? = null

    /**
     * whether detected or undetected
     */
    @Column(name = "occurrence_status")
    var occurrenceStatus: String? = null

    @Column(name = "comments", columnDefinition = "TEXT")
    var comments: String? = null

    @Column(name = "identification_method",columnDefinition = "TEXT")
    var identificationMethod: String? = null

    @Column(name = "data_collection_method")
    var dataCollectionMethod: String? = null

    @Column(name = "visually_identified")
    var visuallyIdentified: Boolean? = null

    @Column(name = "image_url")
    var imageUrl: String? = null

    @Column(name = "identified_by")
    var identifiedBy: String? = null
}